src/posthog-featureflags.ts

```diff
@@ -287,6 +302,46 @@ export class PostHogFeatureFlags {
         return () => this.removeFeatureFlagsHandler(callback)
     }
 
+    updateFeaturePreviewEnrollment(key: string, isEnrolled: boolean): void {
+        this.instance.capture('$feature_enrollment_update', {
+            $feature_flag: key,
+            $feature_enrollment: isEnrolled,
+            $set: {
+                [`$feature_enrollment/${key}`]: isEnrolled,
+            },
+        })
+        // TODO: register in setPersonPropertiesForFlags
+        const newFlags = { ...this.getFlagVariants(), [key]: isEnrolled }
+        this.instance.persistence.register({
+            [PERSISTENCE_ACTIVE_FEATURE_FLAGS]: Object.keys(filterActiveFeatureFlags(newFlags)),
+            [ENABLED_FEATURE_FLAGS]: newFlags,
+        })
+        this._fireFeatureFlagsCallbacks()
+    }
+
+    getFeaturePreviews(force_reload = false): Promise<FeaturePreview[]> {
```

Comment:
@EDsCODE I figured it made more sense to make this on-demand, rather than on library load, because that massively affects resource usage: We'd need to put a lot more into making sure this works if it were called as much as decide, rather than when required.

This decision changes the interface a bit, as I'm returning a promise, rather than the value itself.

Reply:
ah dang it, IE11 compatibility requires callbacks. Will switch back